packages/sil-voting-app/src/store/extension/actions.ts

```diff
@@ -1,11 +1,15 @@
 import { Governor } from "@iov/bns-governance";
 
-import { sendGetIdentitiesRequest } from "../../communication/identities";
+import { GetIdentitiesResponse, sendGetIdentitiesRequest } from "../../communication/identities";
 import { getBnsConnection } from "../../logic/connection";
 import { ExtensionState, SetExtensionStateActionType } from "./reducer";
 
+let identities: GetIdentitiesResponse | undefined;
+
 export async function getExtensionStatus(): Promise<ExtensionState> {
-  const identities = await sendGetIdentitiesRequest();
+  if (!identities) {
+    identities = await sendGetIdentitiesRequest();
+  }
```

Comment:
Why is this cache needed? It feels like the real bug is that getExtensionStatus is called twice

Reply:
This is already cached in redux

Reply:
I agree, but not sure of how to access governor without using `getExtensionStatus()`.

Reply:
After login

```ts
    const result = await getExtensionStatus();
    dispatch(setExtensionStateAction(result.connected, result.installed, result.governor));
```

it should be available in your store:

<img width="900" alt="Bildschirmfoto 2019-08-08 um 14 05 34" src="https://user-images.githubusercontent.com/2603011/62701880-a8abee00-b9e5-11e9-8fd1-f2cb252121ec.png">

However, I don't really know how to access this store. Maybe @albertandrejev can help

Reply:
Ok, we are on the same page then, I'll try that approach.

Reply:
Surely you can just access the store in the component and pass the governor to the `getProposals` function?

Reply:
I found the simplest solution to be `export const store`, but that seemingly may have some consequences that I don't fully understand. So I implemented the solution with a Redux Middleware that saves the governor to a variable. However, I would need some advice on the `if (!governor)` checks in `packages/sil-voting-app/src/store/proposals/actions.ts`.

Reply:
Also you can use `Thunk` middleware, it will allow you to access `state` as well as `dispatch`